pip/utils/packaging.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import absolute_import
+import logging
+import sys
+
+from pip._vendor.packaging import specifiers
+from pip._vendor.packaging import version
+
+logger = logging.getLogger(__name__)
+
+
+def check_requires_python(requires_python):
+    """
+    Check if the python version in used match the `requires_python` specifier.
+
+    Return `True` if the version of python in use matches the requirement.
+    Return `False` if the version of python in use does not matches the
+    requirement. Raises an InvalidSpecifier if `requires_python` have an
+    invalid format.
+    """
+    if requires_python is None:
+        # The package provides no information
+        return True
+    try:
+        requires_python_specifier = specifiers.SpecifierSet(requires_python)
+    except specifiers.InvalidSpecifier as e:
+        logger.debug(
+            "Package %s has an invalid Requires-Python entry - %s" % (
```

Comment:
We don't know the package name here, so this log will be confusing. I'd leave the error uncaught here and log it where we're already catching it in `_link_package_versions()`.